bundler/views.py

```diff
@@ -31,17 +34,32 @@ def build(request, pk):
 
     if b is None or b.done:
         b = BundleTaskModel.create(c)
-        bundle.delay(b)
+        result = bundle.delay(b.pk)
+        b.current_task_id = result.task_id
+        b.save()
     else:
         raise errors.HTTP400Exception('wizard/challenge/error.html', 'build is in progress',
                                       """The bundle is already being built.""",
                                       challenge=c)
-
     return redirect(reverse('wizard:challenge:build',
                             kwargs={'pk': pk}))
 
 
 @login_required
+def cancel(request, pk):
+    b = get_object_or_404(BundleTaskModel, pk=pk)
+    try:
+        if b.current_task_id is not None:
+            revoke(task_id=b.current_task_id, state=celery.states.REVOKED, signal='SIGKILL', terminate=True)
+            b.state = BundleTaskModel.CANCELLED
+            b.save()
+    except KeyError:
+        print("No result yet")
```

Comment:
Ah! These prints! Are they helpful enough? They should really be `logger.debug` or something..